### PR TITLE
feat: add options to truncate

### PR DESCRIPTION
**Pull request by peterdeme**
added truncate options

### CODE REVIEW

**Comment by driver-devel** [comment]
We're moving public class ChannelTruncateRequest into another namespace, so it's breaking change. We could make builder from this object.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	./gradlew :spotlessApply
+
+test:
+	./gradlew test
+
+test_with_docker:
+	docker run -t -i --rm -w /code -v $(PWD):/code --env-file .env openjdk:15 sh -c "sh ./gradlew test"
+
+format_with_docker:
+	docker run -t -i --rm -w /code -v $(PWD):/code openjdk:15 sh -c "sh ./gradlew :spotlessApply"
```
Comment (driver-devel) [inline Makefile:14]: Let's add LF at the end of file

**Comment by driver-devel** [comment]
@peterdeme feel free to ignore breaking change since this not widely used i think we can proceed with that

**Comment by ferhatelmas** [comment]
I bumped rate limits of the app 20x

File: src/test/java/io/getstream/chat/java/ChannelTest.java [hunk 2]
```diff
@@ -121,6 +122,27 @@ void whenTruncateChannel_thenNoException() {
...
+    ChannelGetResponse channelGetResponse =
+        Assertions.assertDoesNotThrow(() -> createRandomChannel());
+    MessageRequestObject message =
+        MessageRequestObject.builder()
+            .text("Truncate channel")
+            .userId(testUserRequestObject.getId())
+            .build();
+    Assertions.assertDoesNotThrow(
+        () ->
+            Channel.truncate(
```
Comment (driver-devel) [inline src/test/java/io/getstream/chat/java/ChannelTest.java:138]: Can we check that we have message in the response ?

Comment (peterdeme) [reply src/test/java/io/getstream/chat/java/ChannelTest.java]: ya sure

File: Makefile (hunk 1 above)
Comment (peterdeme) [reply Makefile]: Added

**Review by driver-devel — APPROVED** [review]
🚀

**Pull request merged**